### PR TITLE
Docs: Add override policy doc

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Added a draft document for override policy.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Only focus on new-added `CommandOverrider` and `ArgsOverrider`.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

cc @lfbear @betaincao @Tingtal

### CODE REVIEW

**Review by Tingtal — COMMENTED** [review]
第15-18行，vary一词一般比较少用作及物动词，即一般会说sth varies from A to B，比较少见vary sth。类似意思的词可以用convert、change、overwrite、override

File: docs/override_policy.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+- `PlaintextOverrider`: a general-purpose tool to varying any kind of resources.
+
+### ImageOverrider
+TBD
+### CommandOverrider
+The `CommandOverrider` is a refined tool to override commands(e.g.`/spec/template/spec/containers/0/command`) 
+for workloads, such as `Deployment`.
+
+The allowed operations are as follows:
+- `add`: append one or more flags to the command list.
```
Comment (Tingtal) [inline docs/override_policy.md:27]: append加个s

File: docs/override_policy.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+## Difference between OverridePolicy and ClusterOverridePolicy
+TBD
+## Resource Selector
+TBD
+## Target Cluster
+TBD
+## Overriders
+
+Karmada offers various alternatives to declare the override rules:
+- `ImageOverrider`: dedicated to varying images for workloads.
```
Comment (Tingtal) [inline docs/override_policy.md:15]: 第15行，vary一词一般比较少用作及物动词，即一般会说sth varies from A to B，比较少见vary sth。类似意思的词可以用convert、change、overwrite、override

File: docs/override_policy.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+### CommandOverrider
+The `CommandOverrider` is a refined tool to override commands(e.g.`/spec/template/spec/containers/0/command`) 
+for workloads, such as `Deployment`.
+
+The allowed operations are as follows:
+- `add`: append one or more flags to the command list.
+- `remove`: removes one or more flags from the command list.
+
+#### Examples
+Suppose we have a deployment named `myapp` as follows.
```
Comment (Tingtal) [inline docs/override_policy.md:31]: Create a Deployment named 'myapp'.

File: docs/override_policy.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      containers:
+        - image: myapp
+          name: myapp
+          command:
+            - ./myapp
+            - --parameter1=foo
+            - --parameter2=bar
+```
+
+**Example 1: Add flags when workloads propagating to specific clusters.**
```
Comment (Tingtal) [inline docs/override_policy.md:50]: **Example 1: Add flags when workloads are propagating to specific clusters.**

File: docs/override_policy.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+spec:
+  ...
+  overriders:
+    commandOverrider:
+      - containerName: alpine
+        operator: add
+        value:
+          - --cluster=member1
+```
+It means `add`(append) a new flag `--cluster=member1` to the `myapp`.
```
Comment (Tingtal) [inline docs/override_policy.md:65]: It means `add` (appending) a new flag `--cluster=member1` to `myapp`.

File: docs/override_policy.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        - image: myapp
+          name: myapp
+          command:
+            - ./myapp
+            - --parameter1=foo
+            - --parameter2=bar
+            - --cluster=member1
+```
+
+**Example 2: Remove flags when workloads propagating to specific clusters.**
```
Comment (Tingtal) [inline docs/override_policy.md:79]: **Example 2: Remove flags when workloads are propagating to specific clusters.**

File: docs/override_policy.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+Karmada offers various alternatives to declare the override rules:
+- `ImageOverrider`: dedicated to varying images for workloads.
+- `CommandOverrider`: dedicated to varying commands for workloads.
+- `ArgsOverrider`: dedicated to varying args for workloads.
+- `PlaintextOverrider`: a general-purpose tool to varying any kind of resources.
+
+### ImageOverrider
+TBD
+### CommandOverrider
+The `CommandOverrider` is a refined tool to override commands(e.g.`/spec/template/spec/containers/0/command`)
```
Comment (Tingtal) [inline docs/override_policy.md:23]: CommandOverrider是不是应该小写c，commandOverrider

File: docs/override_policy.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+      containers:
+        - image: myapp
+          name: myapp
+          command:
+            - ./myapp
+            - --parameter2=bar
+```
+
+### ArgsOverrider
+The `ArgsOverrider` is a refined tool to override args(e.g.`/spec/template/spec/containers/0/args`) for workloads,
```
Comment (Tingtal) [inline docs/override_policy.md:107]: The `ArgsOverrider` is a refined tool to override args (such as `/spec/template/spec/containers/0/args`) for workloads,

File: docs/override_policy.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+such as `Deployment`.
```
Comment (Tingtal) [inline docs/override_policy.md:108]: such as `Deployments`.

File: docs/override_policy.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+The allowed operations are as follows:
+- `add`: append one or more args to the command list.
```
Comment (Tingtal) [inline docs/override_policy.md:111]: appends

File: docs/override_policy.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+- `remove`: removes one or more args from the command list.
+
+Note: the usage of `ArgsOverrider` is similar to `CommandOverrider`, please refer to `CommandOverrider` for examples.
```
Comment (Tingtal) [inline docs/override_policy.md:114]: Note: The use of `ArgsOverrider` is similar to `CommandOverrider`. You can refer to the `CommandOverrider` examples.

**Comment by RainbowMango** [comment]
All comments are addressed. 
@lfbear @betaincao Any comments from you?

**Comment by betaincao** [comment]
> All comments are addressed.
> @lfbear @betaincao Any comments from you?

That looks good to me.

File: docs/override_policy.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: OverridePolicy
+metadata:
+  name: example
+spec:
+  ...
+  overriders:
+    commandOverrider:
+      - containerName: alpine
```
Comment (lfbear) [inline docs/override_policy.md:60]: Maybe the `containerName` should be `myapp` not `alpine`

File: docs/override_policy.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
```
Comment (lfbear) [inline docs/override_policy.md:89]: ditto

**Review by lfbear — CHANGES_REQUESTED** [review]

**Comment by lfbear** [comment]
Hah, there is a typo to be fixed

File: docs/override_policy.md (hunk 12 above)
Comment (RainbowMango) [reply docs/override_policy.md]: Done

**Pull request merged**